components/brave_today/browser/publishers_controller.cc

```diff
@@ -103,6 +113,47 @@ void PublishersController::EnsurePublishersIsUpdating() {
                     << publisher_id;
           }
         }
+        // Add direct publishers (rss feeds)
+        const base::DictionaryValue* direct_feeds_pref =
+            controller->prefs_->GetDictionary(prefs::kBraveTodayDirectFeeds);
+        for (auto const kv : direct_feeds_pref->DictItems()) {
+          if (!kv.second.is_dict()) {
+            // Handle unknown value type
+            LOG(ERROR) << "Found unknown dictionary pref value for"
+                          "Brave News direct feeds at the pref path: "
+                       << prefs::kBraveTodayDirectFeeds << " > " << kv.first;
+            // TODO(petemill): delete item from pref dict?
+            continue;
+          }
+          const base::DictionaryValue& value(
+              base::Value::AsDictionaryValue(kv.second));
+          VLOG(1) << "Found direct feed in prefs: " << kv.first;
+          auto publisher = mojom::Publisher::New();
+          publisher->feed_source = GURL(
+              *value.FindStringKey(prefs::kBraveTodayDirectFeedsKeySource));
+          publisher->publisher_id = kv.first;
+          publisher->publisher_name =
+              *value.FindStringKey(prefs::kBraveTodayDirectFeedsKeyTitle);
+          publisher->type = mojom::PublisherType::DIRECT_SOURCE;
+          // This is always true for direct feeds, reserved property for
+          // "combined source" feeds, and perhaps marking a direct feed
+          // as "bad".
+          publisher->is_enabled = true;
+          // TODO(petemill): Allow the user to disable but not delete a feed
+          publisher->user_enabled_status = mojom::UserEnabled::NOT_MODIFIED;
+          // Validate
+          if (!publisher->feed_source.is_valid()) {
```

Comment:
I think we can do this validation more earlier like before creating `publisher` instance.

Reply:
you're right, I thought I would be validating more things. I've moved it up higher in https://github.com/brave/brave-core/compare/f4f1b4d6eb6fa0acbfb1c93897d26ac6f265e948..7aec98389bea9a3bb41d706631a13d9f3a7fe1fb